backend/dataall/core/resource_lock/db/resource_lock_repositories.py

```diff
@@ -0,0 +1,122 @@
+import logging
+
+from dataall.core.resource_lock.db.resource_lock_models import ResourceLock
+from sqlalchemy import and_, or_
+
+log = logging.getLogger(__name__)
+
+
+class ResourceLockRepository:
+    @staticmethod
+    def create_resource_lock(
+        session, resource_uri, resource_type, is_locked=False, acquired_by_uri=None, acquired_by_type=None
+    ):
+        resource_lock = ResourceLock(
+            resourceUri=resource_uri,
+            resourceType=resource_type,
+            isLocked=is_locked,
+            acquiredByUri=acquired_by_uri,
+            acquiredByType=acquired_by_type,
+        )
+        session.add(resource_lock)
+        session.commit()
+
+    @staticmethod
+    def delete_resource_lock(session, resource_uri):
+        resource_lock = session.query(ResourceLock).filter(ResourceLock.resourceUri == resource_uri).first()
+        session.delete(resource_lock)
+        session.commit()
+
+    @staticmethod
+    def acquire_locks(resources, session, acquired_by_uri, acquired_by_type):
```

Comment:
should this be a [ContextManager](https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager)?

Reply:
I changed logic to make `acquire_lock_with_retry(...)` as a context manager which will acquire on enter and release on exit

Let me know if this is what you were thinking as well

Reply:
I think `acquire_locks` and `release_lock` should be private methods and as such must be prefixed with `_` and optionally moved to the bottom of the file.
If we want to expose acquire/release without a retry we should write another contextmanager.